osgi/impl/src/main/java/org/pentaho/ctools/cde/cache/impl/Cache.java

```diff
@@ -35,39 +42,51 @@
 public final class Cache implements ICache {
   private static final String CACHE_CFG_FILE = "ehcache.xml";
   private static final String CACHE_NAME = "pentaho-cde";
-  private final Ehcache ehcache;
+  private static final String ENABLE_SHUTDOWN_HOOK_PROPERTY = "javax.cache.CacheManager.enableShutdownHook";
+  private final javax.cache.Cache ehcache;
 
   public Cache( IReadAccess readAccess ) throws InitializationException {
     CacheManager cacheManager;
     try {
-      cacheManager = CacheManager.create( readAccess.getFileInputStream( CACHE_CFG_FILE ) );
+      CachingProvider cachingProvider = Caching.getCachingProvider();
+      URI configUri = getClass().getClassLoader().getResource( CACHE_CFG_FILE ).toURI();
+      cacheManager = cachingProvider.getCacheManager( configUri, getClass().getClassLoader() );
+      InputStream inputStream = readAccess.getFileInputStream( CACHE_CFG_FILE );
```

Comment:
I don't understand the purpose of this 'inputStream' variable...
Besides that, this should be a try-with-resources so that we don't leak it...

Reply:
In the master, in order to read the configuration file we were using InputStream and provide the same to CacheManager.  Here if there is any problem with the configuration file, it use to throw an error. Now this check is added to check if the configuration file exists and readable. 

However, I have changed the logic to create a CacheManager. Removed usage of InputStream.